src/funk/fd_funk_rec.c

```diff
@@ -67,7 +68,12 @@ fd_funk_rec_query_global( fd_funk_t *               funk,
     do {
       fd_funk_xid_key_pair_t pair[1]; fd_funk_xid_key_pair_init( pair, fd_funk_txn_xid( txn ), key );
       fd_funk_rec_t const * rec = fd_funk_rec_map_query_const( rec_map, pair, NULL );
-      if( FD_LIKELY( rec ) ) return rec;
+      if( FD_LIKELY( rec ) ) {
+        if( FD_UNLIKELY(NULL != txn_out  ) ) {
+          *txn_out = txn;
+        }
+        return rec;
+      }
       txn = fd_funk_txn_parent( (fd_funk_txn_t *)txn, txn_map );
```

Comment:
nit: let's not overwrite txn here, lets make a new variable

Reply:
This was the previous behavior